src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -347,5 +347,127 @@ private static bool IsMultidimensionalInitializer(ImmutableArray<BoundExpression
 
             return inits.Length != 0 && inits[0].Kind == BoundKind.ArrayInitialization;
         }
+
+        private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExpression wrappedExpression, bool used, bool inPlace)
+        {
+            ImmutableArray<byte> data = default;
+            int elementCount = -1;
+            TypeSymbol elementType = null;
+
+            if (!_module.SupportsPrivateImplClass)
+            {
+                return false;
+            }
+
+            var ctor = ((MethodSymbol)this._module.Compilation.GetWellKnownTypeMember(WellKnownMember.System_ReadOnlySpan_T__ctor));
+            if (ctor == null)
+            {
+                return false;
+            }
+
+            if (wrappedExpression is BoundArrayCreation ac)
+            {
+                var arrayType = (ArrayTypeSymbol)ac.Type;
+                elementType = arrayType.ElementType.EnumUnderlyingType();
+
+                // NB: we cannot use this approach for element types larger than one byte
+                //     the issue is that metadata stores blobs in little-endian format
+                //     so anything that is larger than one byte will be incorrect on a big-endian machine
+                //     With additional runtime support it might be possible, but not yet.
+                //     See: https://github.com/dotnet/corefx/issues/26948 for more details
+                if (elementType.SpecialType.SizeInBytes() != 1)
+                {
+                    return false;
+                }
+
+                elementCount = TryGetRawDataForArrayInit(ac.InitializerOpt, out data);
+            }
+
+            if (elementCount < 0)
+            { 
+                return false;
+            }
+
+            if (!inPlace && !used)
```

Comment:
Why is this being tested so late? Seems like it should happen [way up here](https://github.com/dotnet/roslyn/pull/24621/files#diff-1e1b32781086289e35025f3ef987f012bfe6f3fec3a25312e5f3566a977e9d98R357)

Reply:
If you believe this is a problem, feel free to open a new issue which references an up-to-date version of the source, and includes a scenario where an incorrect or undesirable behavior occurs because of this. Thanks!